src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -421,7 +421,7 @@ public bool ContainsDiagnostics
         /// </summary>
         public bool Contains(SyntaxNode node)
         {
-            if (node == null || !node.FullSpan.IntersectsWith(node.FullSpan))
+            if (node == null || !this.FullSpan.IntersectsWith(node.FullSpan))
```

Comment:
Also it seems that Contains implies that node "is or a child of" this. If you compare trees, it would seem either could be a parent of another.